sheetfu/modules/table.py

```diff
@@ -112,8 +112,39 @@ def sort(self, field, reverse=False):
         if not self.items_range:
             return
         self.items.sort(key=lambda item: item.get_field_value(field), reverse=reverse)
-        for index, item in enumerate(self.items):
-            item.row_index = index
+        self._recalculate_item_indexes()
+        self._generate_full_items_range_batches()
+
+    def delete(self, instances_to_delete):
+        indexes_to_delete = set()
+        if type(instances_to_delete) is not list:
+            instances_to_delete = [instances_to_delete]
+
+        for instance in instances_to_delete:
+            if isinstance(instance, Item):
+                indexes_to_delete.add(instance.row_index)
+            elif type(instance) is int or type(instance) is float:
+                indexes_to_delete.add(instance)
+            else:
+                raise ValueError("Specified unknown value to delete (" + repr(instance) +
+                                 "). Please specify either an index or an Item instance (or a list of these).")
+
+        for index in sorted(indexes_to_delete, reverse=True):
+            if index >= len(self.items):
```

Comment:
Why not catch the exception (Index Error) that `del`would throw when deleting and out of bouds item?

Reply:
The purpose here was not to catch the error (something should be thrown in this case), but rather to provide a more helpful exception message than the one python would throw

Reply:
You could still do it inside the catch clause. Just catch the exception and then throw your custom one.

Reply:
I guess it comes down to personal coding style preference, I've never been a fan of catching exceptions to re-raise my own if know exactly what is going to cause the exception xD

Let's have @philippe2803 un-tie this =P

Reply:
Personal preference for Alex view on this.

Reply:
I concede! 👍